lib/datadog/appsec/event.rb

```diff
@@ -104,8 +105,15 @@ def self.build_service_entry_tags(event_group)
           tags['_dd.origin'] = 'appsec'
 
           # accumulate triggers
+          waf_result = event[:waf_result]
           tags['_dd.appsec.triggers'] ||= []
-          tags['_dd.appsec.triggers'] += event[:waf_result].events
+          tags['_dd.appsec.triggers'] += waf_result.events
+
+          waf_result.derivatives.each do |key, value|
+            tags[key] = JSON.dump(value)
+          end
```

Comment:
Hmm I think I get it now, but coming back to this after a long time I feel some confusion.